spec/acceptance/group/group_local_user_sid_spec.rb

```diff
@@ -77,37 +77,30 @@
     let(:verify_group_command) { "icacls #{target}" }
     let(:group_regex) { %r{.*\\tom:\(M\)} }
 
-    windows_agents.each do |agent|
-      context "on #{agent}" do
-        it 'applies setup manifest' do
-          execute_manifest_on(agent, setup_manifest, debug: true) do |result|
-            expect(result.stderr).not_to match(%r{Error:})
-          end
-        end
-
-        it 'retrieves SID of user account' do
-          on(agent, get_group_sid_command) do |result|
-            sid = sid_regex.match(result.stdout)[1]
-          end
-        end
-
-        it 'applies ACL manifest' do
-          execute_manifest_on(agent, acl_manifest, debug: true) do |result|
-            expect(result.stderr).not_to match(%r{Error:})
-          end
-        end
-
-        it 'verifies ACL rights' do
-          on(agent, verify_group_command) do |result|
-            expect(result.stdout).to match(%r{#{group_regex}})
-          end
-        end
-
-        it 'verifies file data integrity' do
-          expect(file(verify_content_path)).to be_file
-          expect(file(verify_content_path).content).to match(%r{#{file_content}})
-        end
+    it 'applies setup manifest' do
+      idempotent_apply(setup_manifest)
+    end
+
+    it 'retrieves SID of user account' do
+      run_shell(get_group_sid_command) do |result|
+        sid = sid_regex.match(result.stdout)[1]
+      end
+    end
+
+    it 'applies ACL manifest' do
+      # TODO: find out why this is not idempotent
```

Comment:
check this out ;)

Reply:
Initially I thought it could be that Puppet was setting the permissions, and fighting Windows but the puppet config setting "manage_internal_file_permissions" is still set to "false".

Reply:
I've created a bug https://tickets.puppetlabs.com/browse/MODULES-9699 to track this issue.